File: 300baicode/Py/08.py
a, b, c = map(int, input().split())
p = (a + b + c) / 2
area = (p * (p - a) * (p - b) * (p - c)) ** 0.5
r = a * b * c / (4 * area)
print(f'{r:.3f}')

File: 300baicode/08.cpp
#include <bits/stdc++.h>
#define FOR(i, a, b) for (int i = a; i <= b; i++)
#define all(x) x.begin(), x.end()

using namespace std;
using ll = long long;

void fastIO() {
    #ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    #endif
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr); 
}

int main() {
    fastIO();
    double a, b, c; cin >> a >> b >> c;
    double semi_perimeter = (a + b + c) / 2;
    double area = sqrt(semi_perimeter * (semi_perimeter - a) * (semi_perimeter - b) * (semi_perimeter - c));
    cout << fixed << setprecision(3) << a * b * c / (4 * area);
}